vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt

```diff
@@ -144,22 +139,12 @@ class RoomDetailViewModel @AssistedInject constructor(
     private var prepareToEncrypt: Async<Unit> = Uninitialized
 
     @AssistedFactory
-    interface Factory {
-        fun create(initialState: RoomDetailViewState): RoomDetailViewModel
+    interface Factory : MavericksAssistedViewModelFactory<RoomDetailViewModel, RoomDetailViewState> {
+        override fun create(initialState: RoomDetailViewState): RoomDetailViewModel
     }
 
-    /**
-     * Can't use the hiltMaverick here because some dependencies are injected here and in fragment but they don't share the graph.
-     */
-    companion object : MavericksViewModelFactory<RoomDetailViewModel, RoomDetailViewState> {
-
+    companion object : MavericksViewModelFactory<RoomDetailViewModel, RoomDetailViewState> by hiltMavericksViewModelFactory() {
         const val PAGINATION_COUNT = 50
-
-        @JvmStatic
-        override fun create(viewModelContext: ViewModelContext, state: RoomDetailViewState): RoomDetailViewModel {
```

Comment:
this workaround is needed specifically because of the `ActivityScoped` voice helpers, have removed from here and added to the `MessageComposer` where it's needed instead